src/Microsoft.DotNet.Build.Tasks/UpdatePackageDependencyVersion.cs

```diff
@@ -22,7 +22,7 @@ public override bool VisitPackage(JProperty package, string projectJsonPath)
             string dependencyVersion;
             if (package.Value is JObject)
             {
-                dependencyVersion = package.Value["version"].Value<string>();
+                dependencyVersion = package.Value["version"]?.Value<string>();
```

Comment:
Is this going to cause issues in any other places were someone forgets the version? Should we consider also checking for type=project?